stateMutators.js

```diff
@@ -0,0 +1,111 @@
+const gameState = {
+    basketPos: 50,
+    basketValue: 0,
+    score: 0,
+    livesRemaining: 3,
+    gameLevel: 1,
+    gameMode: 'running',
+    fallingObjects: [],
+}
+
+const setBasketPos = (state, value) => ({
+    ...state,
+    basketPos: value,
+});
+
+const moveBasketLeftKey = (state, value) => ({
+    ...state,
+    basketPos: basketPos - value,
+});
+
+const moveBasketRightKey = (state, value) => ({
```

Comment:
Just a naming issue: I don't see a reason for this mutator name to have "Key" at the end. It makes it tightly coupled to the idea of the pressing right key to move right, when the concept of which keys should be pressed is not really relevant here in the state management code. I'd just keep it as "moveBasketRight" and leave the decision of what key triggers this mutation up to the event handling logic. Same with the prev function

Reply:
The use of 'Key' was to distinguish between the basket being moved by the mouse or when a key is pressed. I'll update it